### ISSUE TITLE
Helm chart: cannot login with admin credentials

### ISSUE THREAD

**Issue by captjt** [zalando/spilo#188]
It seems that there isn't a possible way to log into postgres with the "admin" user defined in the Kubernetes helm chart [here](https://github.com/kubernetes/charts/blob/master/incubator/patroni/templates/statefulset-patroni.yaml#L37).

By default here are the users once I log in with `-U postgres` password `tea`.
```
postgres=# \du
                                           List of roles
 Role name  |                         Attributes                         |        Member of
------------+------------------------------------------------------------+-------------------------
 admin      | Create role, Create DB, Cannot login                       | {}
 pgq_admin  | Cannot login                                               | {pgq_reader,pgq_writer}
 pgq_reader | Cannot login                                               | {}
 pgq_writer | Cannot login                                               | {}
 postgres   | Superuser, Create role, Create DB, Replication, Bypass RLS | {}
 robot_zmon | Cannot login                                               | {}
 standby    | Replication                                                | {}
```

It doesn't seem like there is an environment variable for `PGPASSWORD_ADMIN` either in the [docs](https://github.com/zalando/spilo/blob/master/ENVIRONMENT.rst). Anywhere I can help with this issue I'd gladly open a PR as well.

**Comment by ozgune** [comment]
I also ran into this issue when I followed the instructions in https://github.com/kubernetes/charts/tree/master/incubator/patroni

The repo says that the admin password is `cola`. However, when I try to log in as an admin user, I get the following.

<pre>
root@ubuntu:/# psql -U admin -h my-release-patroni.default.svc.cluster.local postgres
Password for user admin: 
psql: FATAL:  password authentication failed for user "admin"
FATAL:  password authentication failed for user "admin"
</pre>

The user `postgres` and password `tea` works.

**Comment by captjt** [comment]
Yeah @ozgune the configurations seem to include the `admin` user, but from what I could find it doesn't seem to actually bootstrap the configurations like [this](https://github.com/zalando/patroni/blob/master/postgres0.yml#L57) would do.

**Comment by CyberDem0n** [comment]
Hi @captjt, hi @ozgune!

You are guys both right, there was such configuration section existing in Spilo, but some time ago we [removed](https://github.com/zalando/spilo/commit/cdd6fe2a513e5f2e0da0185c90bb403aadf079ab) admin user from th config. Actually not really removed, but converted it into role: https://github.com/zalando/spilo/blob/master/postgres-appliance/post_init.sh#L7. The idea is that later we grant admin role to some real (human) users.
Obviously when we were doing that, we absolutely forgot about helm chart :(
I am fine with fixing it, but there is one small requirement: `users:` section should appear in configuration file only if `PGPASSWORD_ADMIN` is supplied from outside.

P.S. it seems that you are really interested in PostgreSQL on Kubernetes topic.
May be you should try to test this PR: https://github.com/zalando/patroni/pull/500
It makes it possible to deploy Patroni on K8S without Etcd, ZooKeeper or Consul. Patroni will use  config-maps and Kubernetes API for leader elections and keeping cluster state.

It is very easy to try it:
1. git checkout feature/k8s
2. cd kubernetes
3. docker build -t patroni .
4. docker push patroni
5. kubectl apply -f patroni_k8s.yaml

Off course, you probably need to use a different name and tag for a docker image you are building/using and you should change this name in kubernetes manifest file accordingly: https://github.com/zalando/patroni/pull/500/files#diff-43ef81d288d3700ec887f191a2292b69R19

**Comment by captjt** [comment]
> May be you should try to test this PR: zalando/patroni#500

Wow, that's actually an awesome feature moving forward. Let external dependencies like etcd/zookeeper/etc. the better for running PostgreSQL. I'll definitely check it out. +1 to the whole team @CyberDem0n!

**Comment by joaogbcravo** [comment]
Hey! Not sure I should open a new issue, or just wait for this to be reopen if it worth it. 

**tl;dr; I'm having problems login with admin, but only after the master role changes.**

Using the Helm chart I created a normal cluster and I was able to login with admin:

```
postgres@postgresql-0:/scripts$ psql -U admin postgres
psql (10.4 (Ubuntu 10.4-2.pgdg18.04+1))
Type "help" for help.

postgres=> \du
                                    List of roles
 Role name  |                         Attributes                         | Member of
------------+------------------------------------------------------------+-----------
 admin      | Create role, Create DB                                     | {}
...
```

However, after changing the master role (using patronictl switchover or deleting the master pod for example)

````
postgres@postgresql-3:~$ psql -U admin postgres
psql: FATAL:  role "admin" is not permitted to log in
postgres@postgresql-3:~$ psql -U postgres
psql (10.4 (Ubuntu 10.4-2.pgdg18.04+1))
Type "help" for help.

postgres=# \du
                                    List of roles
 Role name  |                         Attributes                         | Member of
------------+------------------------------------------------------------+-----------
 admin      | Create DB, Cannot login                                    | {}
...
```

I noticed that this happens because of this:
https://github.com/zalando/spilo/blob/master/postgres-appliance/scripts/post_init.sh#L6

Should this be the behaviour? Why give login rights when the cluster bootstraps and lives and as soon as the master changes remove them?


cc: @CyberDem0n

**Comment by CyberDem0n** [comment]
We intentionally converted the admin user to role.
Later it might be granted to the real users: `create user foo in role admin`

**Comment by joaogbcravo** [comment]
Thanks for the fast reply. I would suggest then that from the start the admin role shouldn't have login permissions.
